rust/aptos-indexer-transaction-stream/src/transaction_stream.rs

```diff
@@ -313,93 +309,150 @@ pub async fn create_fetcher_loop(
+        pb_channel_txn_chunk_size: usize,
+    ) -> Self {
+        Self {
+            indexer_grpc_data_service_address,
+            indexer_grpc_http2_ping_interval,
+            indexer_grpc_http2_ping_timeout,
+            indexer_grpc_reconnection_timeout_secs,
             indexer_grpc_response_item_timeout_secs,
-            resp_stream.next(),
+            starting_version,
+            request_ending_version,
+            auth_token,
+            processor_name,
+            transaction_filter,
+            pb_channel_txn_chunk_size,
+            resp_stream: None,
+            connection_id: None,
+            next_version_to_fetch: starting_version,
+            reconnection_retries: 0,
+            last_fetched_version: starting_version as i64 - 1,
+            fetch_ma: MovingAverage::new(3000),
+        }
+    }
+
+    pub async fn init_stream(&mut self) {
+        info!(
+            processor_name = self.processor_name,
+            service_type = PROCESSOR_SERVICE_TYPE,
+            stream_address = self.indexer_grpc_data_service_address.to_string(),
+            start_version = self.starting_version,
+            end_version = self.request_ending_version,
+            "[Parser] Connecting to GRPC stream",
+        );
+        let response = get_stream(
+            self.indexer_grpc_data_service_address.clone(),
+            self.indexer_grpc_http2_ping_interval,
+            self.indexer_grpc_http2_ping_timeout,
+            self.indexer_grpc_reconnection_timeout_secs,
+            self.starting_version,
+            self.request_ending_version,
+            self.auth_token.clone(),
+            self.processor_name.to_string(),
+        )
+        .await;
+        let connection_id = match response.metadata().get(GRPC_CONNECTION_ID) {
+            Some(connection_id) => connection_id.to_str().unwrap().to_string(),
+            None => "".to_string(),
+        };
+        self.connection_id = Some(connection_id);
+        self.resp_stream = Some(response.into_inner());
+        info!(
+            processor_name = self.processor_name,
+            service_type = PROCESSOR_SERVICE_TYPE,
+            stream_address = self.indexer_grpc_data_service_address.to_string(),
+            connection_id = self.connection_id,
+            start_version = self.starting_version,
+            end_version = self.request_ending_version,
+            "[Parser] Successfully connected to GRPC stream",
+        );
+    }
+
+    /// Gets a batch of transactions from the stream. Batch size is set in the grpc server.
+    /// The number of batches depends on our config
+    /// There could be several special scenarios:
+    /// 1. If we lose the connection, we will try reconnecting X times within Y seconds before crashing.
+    /// 2. If we specified an end version and we hit that, we will stop fetching, but we will make sure that
+    ///    all existing transactions are processed
+    ///
+    /// Returns
```

Comment:
Nit, it returns `TransactionStreamOutput` where the `should_continue_fetching` is true if... blah blah.